### PR TITLE
Review request for Microsoft.RecoveryServices to add version stable/2024-02-01

### PR DESCRIPTION
**Pull request by midhunpramod19**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary>Click here to see the details of Step 1, Breaking Changes review</summary>

If you are in purview of Step 1 of the diagram, follow the [Breaking Changes review process].  
**IMPORTANT!** This applies even if you believe your PR was mislabeled, for any reason, including tool failure.
</details>

<details> 
<summary>Click here to see the details of Step 2, ARM review</summary>

See https://aka.ms/azsdk/pr-arm-review.
</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[Breaking Changes review process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by sorabhtandon** [comment]
>             "name": "resourceName",

Similar defenition on ResrouceName has been added in mulitple places can we add in one common place and $ref it ? 

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4972 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by sorabhtandon** [comment]
>             "name": "resourceName",

Same as below, add  a common defenition of resourceName and $ref it

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4622 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by sorabhtandon** [comment]
>             "description": "Automation error response describing why the operation failed.",

Is the description intentional ? 

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4651 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by sorabhtandon** [comment]
>             "description": "Automation error response describing why the operation failed.",

same as above

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4651 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by sorabhtandon** [comment]
>             "name": "resourceName",

same as above

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4683 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by midhunpramod19** [comment]
> > ```
> >         "name": "resourceName",
> > ```
> 
> Similar defenition on ResrouceName has been added in mulitple places can we add in one common place and $ref it ?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4972 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

We have added a pattern to the resource names for newly added APIs. We were getting linting error if we did not add it. We already have a reference below, if we add the pattern in that, it will be a breaking change

**Comment by sorabhtandon** [comment]
>             "description": "Automation error response describing why the operation failed.",

Is the description message intentional ? 

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4715 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by midhunpramod19** [comment]
> > ```
> >         "description": "Automation error response describing why the operation failed.",
> > ```
> 
> Is the description intentional ?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4651 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

Yes, this is an existing message for other APIs also

**Comment by dhruvesh** [comment]
Swagger LintDiff checks are failing. Please check why Checks are failing

**Comment by midhunpramod19** [comment]
> Swagger LintDiff checks are failing. Please check why Checks are failing

We have added new class which are inheriting an existing class with has a property called "AdditionalProperties". So as this is an existing class, we cannot change this and require exception for the same.

**Comment by krdhruva** [comment]
As your existing api contains resource name filed without any pattern, it will create inconsistency for customer while using same resource name for different api.

---
In reply to: [2011981805](https://github.com/Azure/azure-rest-api-specs/pull/28371#issuecomment-2011981805) [](http://example.com/codeflow?ancestors=2011981805)

**Comment by krdhruva** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/operationresults/{jobId}": {

This should be camel case.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:5035 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/operationresults/{jobId}": {

Do you support multiple `operationResults` type? Can this be top level resource?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:5035 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           "description": "The Id.",

This should be descriptive.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:23949 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           "description": "The name.",

same as above. Pls fix throughout the file.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:23954 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "allowedOperations": {

who defines `allowedOperations` list? If this is known upfront, consider using enum to restrict operations values.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24046 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           "type": "string"

Instead of string, consider using enum.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24830 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "activeLocation": {

This should be defined as enum rather than allowing any string.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24034 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "protectionState": {

This should be defined as enum rather than allowing any string.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24026 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "testFailoverState": {

same as above.


---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24038 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "policyId": {

I assume this will always be arm-id?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24136 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "recoveryContainerId": {

arm-id format?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24083 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "clusterProtectedItemIds": {

arm-id format?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24102 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "areAllClusterNodesRegistered": {

Instead of using bool, pls consider converting it into Enum like `registrationStatus`. This will make it extensible for future scenarios.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24114 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "isSharedDiskVirtualNode": {

Consider converting this to `diskMode` enum to make it extensible.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:23501 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "protectionState": {

Consider adding Enum as string rather than any type of string.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24809 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           }

All of these properties are repeated multiple times across diff types. Consider to moving these to single class and reference them?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24826 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           "description": "The consolidated protection health for the VM taking any issues with SRS as well as all the replication units associated with the VM's replication group into account. This is a string representation of the ProtectionHealth enumeration.",

how is this enforced?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24829 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "clusterRecoveryPointId": {

arm-id format?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:11919 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/examples/ReplicationProtectionClusters_ApplyRecoveryPoint.json [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    "api-version": "2024-02-01",
+    "replicationProtectionClusterName": "testcluster",
+    "protectionContainerName": "pri-cloud-eastus",
+    "fabricName": "fabric-pri-eastus",
+    "resourceName": "vault1",
+    "resourceGroupName": "resourceGroupPS1",
+    "subscriptionId": "7c943c1b-5122-4097-90c8-861411bdd574",
+    "applyClusterRecoveryPointInput": {
+      "properties": {
+        "clusterRecoveryPointId": "/Subscriptions/7c943c1b-5122-4097-90c8-861411bdd574/resourceGroups/resourceGroupPS1/providers/Microsoft.RecoveryServices/vaults/shashankvaultpvt/replicationFabrics/fabric-pri-eastus/replicationProtectionContainers/pri-cloud-eastus/replicationProtectionClusters/testcluster/recoveryPoints/cc48b7f3-b267-432b-ad76-45528974dc62",
```
Comment (krdhruva) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/examples/ReplicationProtectionClusters_ApplyRecoveryPoint.json:12]: Pls add example with `individualNodeRecoveryPoints` details.

**Comment by krdhruva** [comment]
>               "$ref": "#/definitions/ReplicationProtectionCluster"

This does not match with example mentioned in ReplicationProtectionClusters_Create.json

Should this be taking input with `instanceType`?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4768 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           "type": "string"

arm-id format

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:12341 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           "202": {

This api should return 200/204 as final status.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:5278 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>             "description": "Failover input.",

nit: this can be more descriptive.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:5355 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "failoverDirection": {

consider moving this to enum

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:12459 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>       "properties": {

This does not contain `clusterRecoveryPoint` input mentioned in `ReplicationProtectionClusters_TestFailover.json`.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:12530 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationProtectionClusters": {

what is the difference between `replicationProtectionClusters` under vaults vs replicationprotectioncontainers?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:7906 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           "description": "The replication protection cluster Id.",

arm-id format

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:9835 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>         "isClusterInfraReady": {

Consider moving this to enum as `clusterInfraState`.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:10579 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
>           "description": "The test VM details.",

this does not look related to class.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:12308 in 15a2fb1. [](commit_id = 15a2fb1a84b6732681197646eea5f0067459df12, deletion_comment = False)

**Comment by krdhruva** [comment]
Does your RP applies any pattern matching for resource name in current api?

---
In reply to: [2031081086](https://github.com/Azure/azure-rest-api-specs/pull/28371#issuecomment-2031081086) [](http://example.com/codeflow?ancestors=2031081086,2011981805)

**Comment by krdhruva** [comment]
As per offline conversation, RP does not have any pattern matching hence we'll apply suppression for pattern validation linter error.

---
In reply to: [2036280880](https://github.com/Azure/azure-rest-api-specs/pull/28371#issuecomment-2036280880) [](http://example.com/codeflow?ancestors=2036280880,2031081086,2011981805)

**Comment by midhunpramod19** [comment]
> > ```
> >     "allowedOperations": {
> > ```
> 
> who defines `allowedOperations` list? If this is known upfront, consider using enum to restrict operations values.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24046 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

This list can keep changing

**Comment by midhunpramod19** [comment]
> > ```
> >     "activeLocation": {
> > ```
> 
> This should be defined as enum rather than allowing any string.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24034 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

ActiveLocation can be any azure region, we will have to keep updating the swagger when new regions come. So I think it is better for the property to be a string

**Comment by midhunpramod19** [comment]
> > ```
> >     "protectionState": {
> > ```
> 
> This should be defined as enum rather than allowing any string.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:24026 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

These are read only properties, the user does not send these as input

**Comment by midhunpramod19** [comment]
> > ```
> >       "description": "The test VM details.",
> > ```
> 
> this does not look related to class.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:12308 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

Test VM is created during test failover, the description mentions the same

**Comment by midhunpramod19** [comment]
> > ```
> >           "$ref": "#/definitions/ReplicationProtectionCluster"
> > ```
> 
> This does not match with example mentioned in ReplicationProtectionClusters_Create.json
> 
> Should this be taking input with `instanceType`?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:4768 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

The schema and the example are matching, not all properties are required while creating the cluster. Instance type is required in the input.

**Comment by midhunpramod19** [comment]
> > ```
> >     "instanceType": {
> > ```
> 
> Could not find `clusterRecoveryPointId` mentioned in `ReplicationProtectionClusters_UnplannedFailover.json` example.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:12576 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

ClusterRecoveryPointId is there in the A2AClusterUnplannedFailoverInput object

**Comment by midhunpramod19** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/operationresults/{jobId}": {
> > ```
> 
> Do you support multiple `operationResults` type? Can this be top level resource?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-02-01/service.json:5035 in 15a2fb1. [](commit_id = [15a2fb1](https://github.com/Azure/azure-rest-api-specs/commit/15a2fb1a84b6732681197646eea5f0067459df12), deletion_comment = False)

Yes we support multiple operationResults

**Review by mentat9 — APPROVED** [review]

**Pull request merged**